src/features/import-map.js

```diff
@@ -34,26 +30,22 @@ export function mergeImportMap(originalMap, newMap) {
   for (let i in newMap.scopes) {
     originalMap.scopes[i] = newMap.scopes[i];
   }
-  return originalMap;
 }
 
-systemJSPrototype.resolve = function (id, parentUrl) {
-  parentUrl = parentUrl || baseUrl;
-
+systemJSPrototype.prepareImport = function () {
   if (acquiringImportMaps) {
     acquiringImportMaps = false;
-    Array.prototype.forEach.call(document.querySelectorAll('script[type="systemjs-importmap"]'), function (script) {
-      importMapPromise = importMapPromise.then(function (map) {
-        return (script._j || script.src && fetch(script.src).then(function (resp) {return resp.json()}) || Promise.resolve(JSON.parse(script.innerHTML)))
-        .then(function (json) {
-          return mergeImportMap(map, parseImportMap(json, script.src || baseUrl));
-        });
-      });
-    });
+    return Promise.all(Array.prototype.map.call(document.querySelectorAll('script[type="systemjs-importmap"]'), function (script) {
+      return (script._j || script.src && fetch(script.src).then(function (resp) {return resp.json()}) || Promise.resolve(JSON.parse(script.innerHTML)))
+      .then(function (json) {
```

Comment:
Wait is there maybe a bug here -- a race condition?

I think we're applying the import maps in the order that we can load them instead of document order. Right? I'll look into a fix for it.

Reply:
Just fixed this ^^ with a new commit.